src/providers/fireworks-ai/imageGenerate.ts

```diff
@@ -0,0 +1,104 @@
+import { FIREWORKS_AI } from '../../globals';
+import { ErrorResponse, ImageGenerateResponse, ProviderConfig } from '../types';
+import { generateInvalidProviderResponseError } from '../utils';
+import {
+  FireworksAIValidationErrorResponse,
+  FireworksAIErrorResponseTransform,
+  FireworksAIErrorResponse,
+} from './chatComplete';
+
+export const FireworksAIImageGenerateConfig: ProviderConfig = {
+  text_prompts: {
+    param: 'text_prompts',
+    required: true,
+  },
+  model: {
+    param: 'model',
+    required: true,
+    default: 'stable-diffusion-xl-1024-v1-0',
+  },
+  height: {
```

Comment:
`height` and `weight` are not directly available in OpenAI specs. You will have to use size and split it by `x`. Please check the OpenAI docs to compare the params